publisher.rb

```diff
@@ -1,33 +1,55 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
+require_relative 'q_it_errors'
 
 class Publisher
   
   attr_reader :sqs, :sleep_period, :queue_name
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
     @queue_name = options[0]
+    @sleep_period = options[1]
+  end
+
+  def self.validate(options)
+    raise QItArgumentError, "Incorrect number of arguments. Expected 2 got #{options.count}" if options.count != 2
+    
+    queue_name = options[0]
+    sleep_period = options[1]
+    valid_queue_name?(queue_name) && valid_sleep_period?(sleep_period)
   end
 
   # For Standard Queue
   def create_std_q
     sqs.create_queue(queue_name: queue_name)
   end
 
-  def valid_sleep_period?(sleep_period)
+  def self.valid_sleep_period?(sleep_period)
     max_sleep_period = /\A([5-9]|1\d{1}|20)\z/
-    !sleep_period.nil? && sleep_period.match(max_sleep_period)
+    
+    raise QItNullSleepTimeError, "QItNullSleepTimeError :: #{self} Sleep period cannot be nil." if sleep_period.nil?
+    raise QItArgumentError, "QItArgumentError :: #{self} Invalid sleep period #{sleep_period}. Sleep period range is between 5-20 seconds." unless sleep_period.match(max_sleep_period)
```

Comment:
Alway use `match?` with an `if` or `unless`, it returns a boolean rather than a `match` which returns an object.

Also if you really believe that you may get nil, then you could use `sleep_period.to_s.match?(/^([5-9]|1\d{1}|20)$/)` which will also take care of nil.

Reply:
I had used `match?`.. Guess I forgot to add the ? here..